### DISCUSSION TITLE
ESM Js file not generating for page.ts

### DISCUSSION THREAD

**Discussion by mvmshakkir** [serenity-is/Serenity#7293] [Help]
import { LookupType, ParameterDataTypes, ParameterEditorTypes } from "../../ServerTypes/Reports";
import { Decorators, Widget, BaseDialog, Select2 } from '@serenity-is/corelib';
import { ReportRequest, CommonService } from "../../ServerTypes/Modules";
import $ from 'jquery';

declare var Morris: any;

@Decorators.registerClass('Car.Rental.Reports.DynamicReportsDialog')
@Decorators.resizable()
@Decorators.maximizable()
export class DynamicReportsDialog extends BaseDialog<any> {

   
    private areaChart: any;
    private static reportmodel: ReportRequest} while importing import { LookupType, ParameterDataTypes, ParameterEditorTypes } from "../../ServerTypes/Reports"; 
import { ReportRequest, CommonService } from "../../ServerTypes/Modules"; this js fie  not generating but when comment this line it generating js file

**Comment by VictorTomaili** [comment]
default entry points are defined here

https://github.com/serenity-is/Serenity/blob/b0d3107b575d9c0ed9aeb9b3d53e0bb9ff607a0d/packages/tsbuild/dist/index.js#L7

Files should ends with `...page.ts` or `...page.tsx`

If the file is not exist in output directory, the javascript code will not be built successfully. Check for any error while ts build.

**Comment by mvmshakkir** [comment]
The problem occurs when I import server types using the following lines:

import { LookupType, ParameterDataTypes, ParameterEditorTypes } from "../../ServerTypes/Reports";
import { ReportRequest, CommonService } from "../../ServerTypes/Modules";
If I comment out these lines, the .js file is generated. Otherwise, it is not generated.

**Reply by VictorTomaili** [reply]
Check your imported files for any issue (even generated ones)
